client/src/app/services/chart.service.ts

```diff
@@ -569,9 +569,12 @@ export class ChartService {
     console.log(`Loading charts with ${this.currentBarCount} bars`);
 
     // get data and load charts
-    this.api.getQuotes('daily', this.currentBarCount)
+    this.api.getQuotes()
       .subscribe({
-        next: (quotes: Quote[]) => {
+        next: (allQuotes: Quote[]) => {
+          
+          // Slice array to desired length based on window size
+          const quotes = allQuotes.slice(-this.currentBarCount);
```

Comment:
These are date sorted, so trimming hare should remove from the older end of the collection.